tests/conftest.py

```diff
@@ -838,22 +839,11 @@ def run_iterations(
         campaign.add_measurements(rec)
 
 
-def get_dummy_training_data(length: int) -> tuple[pd.DataFrame, pd.DataFrame]:
-    """Create column-less input and target dataframes of specified length."""
-    df = pd.DataFrame(np.empty((length, 0)))
-    return df, df
-
-
-def get_dummy_searchspace() -> SearchSpace:
-    """Create a dummy searchspace whose actual content is irrelevant."""
-    parameters = [NumericalDiscreteParameter(name="test", values=(0, 1))]
-    return SearchSpace.from_product(parameters)
-
-
 def select_recommender(
     meta_recommender: MetaRecommender, training_size: int
 ) -> PureRecommender:
-    """Select a recommender for given training dataset size."""
-    searchspace = get_dummy_searchspace()
-    df_x, df_y = get_dummy_training_data(training_size)
-    return meta_recommender.select_recommender(searchspace, train_x=df_x, train_y=df_y)
+    """Select a recommender for a given training dataset size."""
```

Comment:
Shouldn't we add significantly more tests for the new different uses of the recommend function? I would have expected way more tests to ensure that different combinations of providing/not providing the several objects for the different recommenders works as intended. Or is this already checked?

Reply:
first integration tests for unintended use of `NonPredictiveRecommender` now in place, which already cover part of the "providing/not providing objects" story. More to come in the next PRs ...